docs/dataservice-write-example.md

```diff
@@ -1,213 +1,281 @@
 
-```bash
-# Execute as sudo if necessary
-(sudo) make install
-```
+### <a name="build-sdk-android"></a>Build HERE OLP SDK for C++
 
-### Assemble APK and run it on Android device
+To build the HERE OLP SDK for C++ on Android:
 
-After you complete the `Gradle` project configuration, you can use `gradlew` executable to build and install APK on your Android device:
+1. Set `OLP_SDK_BUILD_EXAMPLES` to `ON`.
+2. Specify the path to the Android NDK toolchain file using the `CMAKE_TOOLCHAIN_FILE` variable.
+3. If you want to build the SDK for a specific Android platform, use the `-DANDROID_PLATFORM CMake` flag, and if you want to build the SDK for a specific Android architecture, use the `-DANDROID_ABI` flag.
+   For more details, see [NDK-specific CMake variables](https://developer.android.com/ndk/guides/cmake#variables).
 
-```bash
-cd examples/dataservice-write/android
-./gradlew assembleDebug
-./gradlew installDebug
-```
+   ```bash
+   mkdir build && cd build
+   cmake .. -DOLP_SDK_BUILD_EXAMPLES=ON -DCMAKE_TOOLCHAIN_FILE=$NDK_ROOT/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a
+   ```
 
-Alternatively, you can use Android Studio IDE by opening the `build/examples/dataservice-write/android/build.gradle` script.
+   The `CMake` command generates a `Gradle` project in the `build/examples/dataservice-write/android` folder.
 
-After you provide your app access key id, secret access key, catalog, and layer information, install and run `dataservice_write_example` APK, the main UI screen displays a message `Publish Successful`. If you encounter an error, please check the device's logcat for the error message.
+4. Install the HERE OLP SDK for C++ libraries into the **sysroot** directory.
 
-> **Note:** you can run `CMake` command directly from the `<olp-sdk-root>/examples/dataservice-write/` folder if you have already built and installed HERE OLP SDK for C++ libraries for Android. Make sure you provide correct path to the `LevelDB` library, and correct `OLP_SDK_HTTP_CLIENT_JAR` parameter in the `CMake` command, invoked by the `build/examples/dataservice-write/android/app/build.gradle` script.
+   ```bash
+   # If necessary, execute as sudo.
+   (sudo) make install
+   ```
 
-## Building and running on iOS
+### <a name="build-and-run-android"></a>Build and Run the APK
+
+To build and run the APK:
+
+1. In the Android Studio IDE, open the `build/examples/dataservice-write/android/build.gradle` script.
+2. Provide your application access key ID, access key secret, catalog HRN, and layer name.
+3. Install and run the `dataservice_write_example` APK.
 
-This example shows how to integrate and use the HERE OLP SDK for C++ for a basic iOS application written in the Objective-C language.
+The main screen displays the following message: "Publish Successful".
 
-### Prerequisites
+If you encounter an error, in your device LogCat, find the error message.
```

Comment:
```suggestion
If you encounter an error, find the error message in your device Logcat.
```
Refer to my previous comment in other PR: https://github.com/heremaps/here-olp-sdk-cpp/pull/529/files#r361130896

Reply:
Deleted